lib/mongoid/factory.rb

```diff
@@ -63,12 +79,34 @@ def build(klass, attributes = nil)
     #
     # @return [ Document ] The instantiated document.
     def from_db(klass, attributes = nil, criteria = nil, selected_fields = nil)
+      execute_from_db(klass, attributes, criteria, selected_fields, execute_callbacks: true)
+    end
+
+    # Execute from_db.
```

Comment:
I am not sure this comment is helpful. This method seems to do much more that just calling `from_db`. Maybe we can add some more detailed explanation what is does.

Reply:
Do you have any suggestions? Truth is there is a lot to put here, but I left it all under the from_db method.

Reply:
If this method is just an implementation and should not be called itself, it's fine to leave it as it is now. Maybe we can make this method private to emphasize it.

Reply:
It is an implementation with an additional parameter added to it. It's only called explicitly within our code and is therefore labeled `@api private`... It didn't make sense to make it actually private since it needed to be called from elsewhere in our code.